### ISSUE TITLE
Suppressed rules should not execute autoCorrect

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#4172]
## Environment
Detekt: 1.14.0 and 1.15.0 (can't upgrade further)
Gradle: 6.7 and 6.9.1
OS: Windows 10

## Expected Behavior
Suppressed rules should not execute autoCorrect.

## Current Behavior
Suppressed rules execute autoCorrect.

## Context
I wrote a custom rule ("CompanyLogging") that changes logging calls that are for the wrong class. For example:
```kotlin
private val LOG = logger<D>()
class C { }
```
where `D` is wrong, it should be a logger for `C`, so the autoCorrect changes it to
```kotlin
private val LOG = logger<C>()
class C { }
```
Now, what happens if I want to keep it `D`, because it makes sense? For example
```kotlin
@Suppress("CompanyLogging") // Using the central class of said library.
private val LOG = logger<RandomImageLoaderInAndroid>()
class RandomImageLoaderDependencyInjectionModule { }
```
The problem is that, while suppressed, the code is changed to 
```kotlin
@Suppress("CompanyLogging") // Using the central class of said library.
private val LOG = logger<RandomImageLoaderDependencyInjectionModule>()
class RandomImageLoaderDependencyInjectionModule { }
```
so every time someone would run autoCorrect they would have to revert this change.

_This prevents automated autoCorrect as well, because it would be committed every time a PR branch is autoCorrected._

**Comment by TWiStErRob** [comment]
No, sorry, I don't have time for this now.

Does this mean that it's up to the author of the autoCorrect code to decide:

- fix always (regardless of autoCorrect flag)
- withAutoCorrect
- withAutoCorrect + check isSuppressedBy

As a "quick fix" we could add the existence of this isSuppressedBy to the docs?

**Comment by BraisGabin** [comment]
Thinking a bit more about this. Do we want to implement this? I think that it shouldn't be implemented on the core of detekt. It should be the rule author taking this decission.

So, the only action we could do on this issue is to decide if we want to support it as authors of formatting.

**Comment by chao2zhang** [comment]
This is probably worth consideration when we evolve `detekt-compiler-plugin` (And eventually move to IR). All auto-correction is limited to ktlint rules as of now.
